src/utils/processOutputInterpreters/trivy-image-output.ts

```diff
@@ -0,0 +1,112 @@
+const LineSplitter = /\r?\n/;
+const dateHeader = /^[-\d]+T[-:.\d]+Z?\s+\033\[\d+m([A-Z]+)\033\[\d+m\s+(.*)$/;
+const borderPattern = /^\+(?:-+\+){6}\s*$/;
+const internalBorderPattern = /^\+(?:(?:-+|\s+)\+){6}\s*$/;
+const rowPattern = /^(?:\|[^|]+){6}\|\s*$/;
+
+const CVEKeys = ['Package', 'VulnerabilityID', 'Severity', 'Title', 'InstalledVersion', 'FixedVersion', 'Description', 'PrimaryURL'];
+const severityRatings: Record<string, number> = {
+  LOW:      1,
+  MEDIUM:   2,
+  HIGH:     3,
+  CRITICAL: 4,
+  UNKNOWN:  5,
+};
+const MaxSeverityRating = Math.max(...Object.values(severityRatings));
+
+type finalVulType = Record<string, string>;
+
+export default class TrivyScanImageOutputCuller {
+  prelimLines: Array<string>;
+  JSONLines: Array<string>;
+  inJSON = false;
+
+  constructor() {
+    this.prelimLines = [];
+    this.JSONLines = [];
+  }
+
+  getRating(key: string) {
+    return key in severityRatings ? severityRatings[key] : MaxSeverityRating;
+  }
+
+  fixLines(lines: Array<string>) {
+    // "key": "value with an escaped \' single quote isn't valid json"
+    return lines.map(line => line.replace(/\\'/g, "'"));
+  }
+
+  addData(data: string): void {
+    if (this.inJSON) {
+      this.JSONLines.push(data.replace(/\\'/g, "'"));
+
+      return;
+    }
+    const lines = data.split(LineSplitter);
+    const jsonStartIndex = lines.indexOf('[');
+
+    if (jsonStartIndex >= 0) {
+      this.prelimLines = this.prelimLines.concat(lines.slice(0, jsonStartIndex));
+      this.inJSON = true;
+      this.JSONLines = this.fixLines(lines.slice(jsonStartIndex));
+    } else {
+      this.prelimLines = this.prelimLines.concat(lines);
+    }
+  }
+
+  getProcessedData() {
+    const prelimLines = this.prelimLines.map(line => line.replace(dateHeader, '$1 $2'));
+
+    if (!this.inJSON) {
+      // No JSON, just so return the lines we have
+      return prelimLines.join('\n');
+    }
+    let core;
+
+    try {
+      core = JSON.parse(this.JSONLines.join(''));
```

Comment:
Us using JSON means that we can only get any result once the whole thing is done, right? At which point we can do something like add a known marker on the last line, just accumulate the text until we see that marker (and hence the last line), and parse the whole thing in one go.

Alternatively, for a better experience, we can instead do line-oriented JSON (that is, a whole json per line, for one entry), and deal with things that way…

Reply:
This would work much better if we can declare a type for the result, so that TypeScript knows what fields we expected, and we don't have to use `[]` indexing for everything.

Reply:
Can we spin this out as a separate issue? As an improvement over the work in this story, rather than part of it?

Reply:
Created #444